docs/cli-arguments.md

```diff
@@ -27,6 +27,7 @@ $ kube-state-metrics -h
 Usage of ./kube-state-metrics:
       --add_dir_header                   If true, adds the file directory to the header
       --alsologtostderr                  log to standard error as well as files
+      --annotation-allowlist string      Comma-separated list of annotation labels to be exposed. This list comprises of exact annotation label names and/or regex patterns.
```

Comment:
I prefer leaving out `allowlist` as a part of the param name since there is allow/deny dichotomy here.

Perhaps something along the lines of `--annotation-filter`?

Thoughts @brancz @lilic ?